src/v2_0/rws_interface.cpp

```diff
@@ -1008,28 +1004,31 @@ IOSignalInfo RWSInterface2::getIOSignals()
 }
 
 
-void RWSInterface2::requestMastership(std::string const& domain)
+void RWSInterface::requestMastership(MastershipDomain domain)
 {
-  rws_client_.requestMastership(domain);
+  std::stringstream uri;
+  uri << "/rw/mastership/" << domain << "/request";
+  rws_client_.httpPost(uri.str(), "", "application/x-www-form-urlencoded;v=2.0");
 }
 
 
-void RWSInterface2::releaseMastership(std::string const& domain)
+void RWSInterface::releaseMastership(MastershipDomain domain)
 {
-  rws_client_.releaseMastership(domain);
+  std::stringstream uri;
+  uri << "/rw/mastership/" << domain << "/release";
+  rws_client_.httpPost(uri.str(), "", "application/x-www-form-urlencoded;v=2.0");
 }
```

Comment:
I think low level logic regarding URIs and content should stay in rws_client, as this is the case with other methods

Reply:
I want to get rid of the RWSClient class.